prow/github/client.go

```diff
@@ -1033,6 +1033,8 @@ func (c *client) requestRetryWithContext(ctx context.Context, method, path, acce
 		} else if errors.Is(err, &appsAuthError{}) {
 			c.logger.WithError(err).Error("Stopping retry due to appsAuthError")
 			return resp, err
+		} else if strings.Contains(err.Error(), "Unable to retry this workflow run because it was created over a month ago") {
+			return resp, err
```

Comment:
Do we still need this now that we have this functionality in the fake client?

Reply:
Didn't get you here, you mean we should allow retry to happen the usual way and let things return in normally cause we have handled the err at caller level i.e. where we call `TriggerFailedGitHubWorkflow`?

Reply:
Oh nvm, I misunderstood this change. I agree with you, we shouldn't retry if its expired 👍🏼 
I would however prefer if this change got a second set of eyes, @petr-muller, could you maybe take a look here? 

If we are modifying how the retry happens, we should have an added test cases for the prow client that shows that for an error message that we are parsing for, we get a `403` and a non-nil error.

You can probably reuse most of this test to achieve that: https://github.com/MadhavJivrajani/test-infra/commit/38c5f900e22c844719e9786e0a0f64b56e705505

Reply:
waiting for final call on this and will update the PR

Reply:
@AvineshTripathi pls feel free to update the PR, its always better to have the changes in front of you for ease of review.